File: WW1 code/Java Classes/EditBurials.java
import java.sql.SQLException;

public class EditBurials {
    private String Surname;
    private String Forename;
    private int DoB;
    private String Medals;
    private String Date_of_Death;
    private String Rank;
    private String Service_Number;
    private String Regiment;
    private String Battalion;
    private String Cemetery;
    private String Grave_Reference;
    private String Info;

    public EditBurials() throws SQLException{}

    public String getSurname() {
        return Surname;
    }

    public void setSurname(String surname) {
        Surname = surname;
    }

    public String getForename() {
        return Forename;
    }

    public void setForename(String forename) {
        Forename = forename;
    }

    public int getDoB() {
        return DoB;
    }

    public void setDoB(int doB) {
        DoB = doB;
    }

    public String getMedals() {
        return Medals;
    }

    public void setMedals(String medals) {
        Medals = medals;
    }

    public String getDate_of_Death() {
        return Date_of_Death;
    }

    public void setDate_of_Death(String date_of_Death) {
        Date_of_Death = date_of_Death;
    }

    public String getRank() {
        return Rank;
    }

    public void setRank(String rank) {
        Rank = rank;
    }

    public String getService_Number() {
        return Service_Number;
    }

    public void setService_Number(String service_Number) {
        Service_Number = service_Number;
    }

    public String getRegiment() {
        return Regiment;
    }

    public void setRegiment(String regiment) {
        Regiment = regiment;
    }

    public String getBattalion() {
        return Battalion;
    }

    public void setBattalion(String battalion) {
        Battalion = battalion;
    }

    public String getCemetery() {
        return Cemetery;
    }

    public void setCemetery(String cemetery) {
        Cemetery = cemetery;
    }

    public String getGrave_Reference() {
        return Grave_Reference;
    }

    public void setGrave_Reference(String grave_Reference) {
        Grave_Reference = grave_Reference;
    }

    public String getInfo() {
        return Info;
    }

    public void setInfo(String info) {
        Info = info;
    }

    EditBurials editBurials = new EditBurials();

    public EditBurials getEditBurials() {
        editBurials.setSurname(Surname);
        editBurials.setForename(Forename);
        editBurials.setDoB(DoB);
        editBurials.setMedals(Medals);
        editBurials.setDate_of_Death(Date_of_Death);
        editBurials.setRank(Rank);
        editBurials.setService_Number(Service_Number);
        editBurials.setRegiment(Regiment);
        editBurials.setBattalion(Battalion);
        editBurials.setCemetery(Cemetery);
        editBurials.setGrave_Reference(Grave_Reference);
        editBurials.setInfo(Info);

        return editBurials;
    }
    EditDAO editDAO = new EditDAO();
    boolean success = editDAO.EditBurials(editBurials);

}

File: EditBurials.php
<?php
class EditBurials {
    private $surname;
    private $forename;
    private $dob;
    private $medals;
    private $date_of_death;
    private $rank;
    private $service_number;
    private $regiment;
    private $battalion;
    private $cemetery;
    private $grave_reference;
    private $info;
    private $original_service_number;

    public function __construct() {
        // Empty constructor
        error_log("EditBurials instance created");
    }

    // Getters and Setters
    public function getSurname() {
        return $this->surname;
    }
    public function setSurname($surname) {
        $this->surname = $surname;
    }

    public function getForename() {
        return $this->forename;
    }
    public function setForename($forename) {
        $this->forename = $forename;
    }

    public function getDoB() {
        return $this->dob;
    }
    public function setDoB($dob) {
        $this->dob = $dob;
    }

    public function getMedals() {
        return $this->medals;
    }
    public function setMedals($medals) {
        $this->medals = $medals;
    }

    public function getDate_of_Death() {
        return $this->date_of_death;
    }
    public function setDate_of_Death($date_of_death) {
        $this->date_of_death = $date_of_death;
    }

    public function getRank() {
        return $this->rank;
    }
    public function setRank($rank) {
        $this->rank = $rank;
    }

    public function getService_Number() {
        return $this->service_number;
    }
    public function setService_Number($service_number) {
        $this->service_number = $service_number;
    }

    public function getRegiment() {
        return $this->regiment;
    }
    public function setRegiment($regiment) {
        $this->regiment = $regiment;
    }

    public function getBattalion() {
        return $this->battalion;
    }
    public function setBattalion($battalion) {
        $this->battalion = $battalion;
    }

    public function getCemetery() {
        return $this->cemetery;
    }
    public function setCemetery($cemetery) {
        $this->cemetery = $cemetery;
    }

    public function getGrave_Reference() {
        return $this->grave_reference;
    }
    public function setGrave_Reference($grave_reference) {
        $this->grave_reference = $grave_reference;
    }

    public function getInfo() {
        return $this->info;
    }
    public function setInfo($info) {
        $this->info = $info;
    }

    public function getOriginal_Service_Number() {
        return $this->original_service_number;
    }
    public function setOriginal_Service_Number($original_service_number) {
        $this->original_service_number = $original_service_number;
        error_log("Original service number set to: " . $original_service_number);
    }

    // Debug function to output all properties
    public function debugOutput() {
        $data = [
            'surname' => $this->surname,
            'forename' => $this->forename,
            'dob' => $this->dob,
            'medals' => $this->medals,
            'date_of_death' => $this->date_of_death,
            'rank' => $this->rank,
            'service_number' => $this->service_number,
            'regiment' => $this->regiment,
            'battalion' => $this->battalion,
            'cemetery' => $this->cemetery,
            'grave_reference' => $this->grave_reference,
            'info' => $this->info,
            'original_service_number' => $this->original_service_number
        ];
        error_log("EditBurials data: " . json_encode($data));
        return $data;
    }

    // Method to save the object via DAO
    public function save($dao = null) {
        if ($dao === null) {
            error_log("Creating new EditDAO instance in EditBurials save method");
            $dao = new EditDAO();
        } else {
            error_log("Using provided EditDAO instance in EditBurials save method");
        }
        
        // Log all data before saving
        error_log("EditBurials save: About to save with data: " . json_encode($this->debugOutput()));
        
        $result = $dao->editBurials($this);
        error_log("EditBurials save result: " . ($result ? "success" : "failure"));
        return $result;
    }
} 